forge/db/models/ProjectSettings.js

```diff
@@ -39,6 +44,18 @@ module.exports = {
     associations: function (M) {
         this.belongsTo(M.Project)
     },
+    finders: function (M) {
+        return {
+            static: {
+                isHostnameUsed: async (hostname) => {
+                    const count = await this.count({
+                        where: { key: KEY_HOSTNAME, value: hostname.toLowerCase() }
+                    })
+                    return count !== 0
+                }
+            }
+        }
```

Comment:
This feels like it's missing something.

What happens if the user enters a hostname with our exisitng domain?

e.g. foo.flowforge.cloud

I think we need a test to see if the supplied domain ends with `app.config.domain` and also fail. This prevents them trying to be clever on the existing system to get round the list of banned names

https://github.com/flowforge/flowforge/blob/01c52ced147ee1c594f0ecc2828a2df0f490df87/forge/routes/api/project.js#L22-L35

Reply:
@hardillb Added in [dc880ee](https://github.com/flowforge/flowforge/pull/1361/commits/dc880ee90979ef990025e6b2c69a2181d1da6c73)